stempy/image.cpp

```diff
@@ -131,28 +179,43 @@ STEMImage createSTEMImage(std::vector<Block>& blocks, int rows, int columns,  in
   auto dark = vtkm::cont::make_ArrayHandle(darkFieldMask, numberOfPixels);
 #endif
 
-  for(const Block &block: blocks) {
-    auto data = block.data.get();
-#ifdef VTKm
-    // Transfer the entire block of data at once.
-    auto dataHandle = vtkm::cont::make_ArrayHandle(
-      data, numberOfPixels * block.header.imagesInBlock);
-#endif
-    for(int i=0; i<block.header.imagesInBlock; i++) {
+  // Run the calculations in a thread pool while the data is read from
+  // the disk in the main thread.
+  int numThreads = 4;
+  ThreadPool pool(numThreads);
+
+  // Populate the worker pool
+  vector<future<void>> futures;
+  for (; first != last; ++first) {
+    Block b = std::move(*first);
```

Comment:
I don't think this move is needed, we are already doing a move @ 196.

Reply:
Okay.